bumpversion/__init__.py

```diff
@@ -336,7 +342,59 @@ def __iter__(self):
         return iter(self._values)
 
     def __repr__(self):
-        return '<bumpversion.Version:{}>'.format(keyvaluestring(self._values))
+        by_part = ", ".join("{}={}".format(k, v) for k, v in self.items())
+        return '<bumpversion.Version:{}>'.format(by_part)
+
+    def __hash__(self):
+        return hash(tuple((k, v) for k, v in self.items()))
+
+    def _compare(self, other, method, strict=True):
+        """
+        For strict relations, if a part is equal we skip it
+        :param other: the other Version
+        :param method: the compare method
+        :return:
+        """
+        try:
+            for vals in ((v, other[k]) for k, v in self.items()):
+                if vals[0] == vals[1]:
+                    continue
+                return method(vals[0], vals[1])
+            return not strict
+        except KeyError:
+            raise TypeError("Versions use different parts, cant compare them.")
+
+    def __eq__(self, other):
+        if self is other:
+            return True
+        if not isinstance(other, Version):
+            return False
+        try:
+            return all(v == other[k] for k, v in self.items())
+        except KeyError:
+            raise TypeError("Versions use different parts, cant compare them.")
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __le__(self, other):
+        return self._compare(other, lambda s, o: o > s, False)  # Note the change of order in operands
```

Comment:
You can also use `operator.lt` instead of the lambda. There's also `operator.gt`